### PR TITLE
Tally Summation for MGXS Subdomain Averaging

### PR DESCRIPTION
**Pull request by wbinventor**
This short PR simply applies the fast `Tally.summation(...)` from #550 to the `MGXS.get_subdomain_avg(...)` method. Fortunately, that method is now largely gutted since it can stand and use the more formal tally summation.

### CODE REVIEW

**Comment by paulromano** [comment]
Can you modify one of the mgxs tests to use this method?

**Comment by wbinventor** [comment]
Sure, I'll add a test for it.

**Comment by wbinventor** [comment]
There is a new "test_mgxs_library_distribcell" test which runs the `MGXS.get_subdomain_avg_xs()` and `Library.get_subdomain_avg_library()` methods. I think this functionality is pretty obscure as I'm likely the only person in the world interested in it at the moment, but I'm happy to include the test to improve the code coverage of the suite.

File: tests/test_mgxs_library_distribcell/test_mgxs_library_distribcell.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                outstr += df.to_string()
+
+        # Hash the results if necessary
+        if hash_output:
+            sha512 = hashlib.sha512()
+            sha512.update(outstr.encode('utf-8'))
+            outstr = sha512.hexdigest()
+
+        return outstr
+
```
Comment (paulromano) [inline tests/test_mgxs_library_distribcell/test_mgxs_library_distribcell.py]: Remove one blank line here

Comment (wbinventor) [inline tests/test_mgxs_library_distribcell/test_mgxs_library_distribcell.py]: Got it

**Pull request merged**